test/lib/users/ssh_authorized_keyring_test.rb

```diff
@@ -90,122 +130,163 @@ def authorized_keys_from_home(path)
   describe "#write_keys" do
     let(:tmpdir) { Dir.mktmpdir }
     let(:home) { File.join(tmpdir, "/home/user") }
-    let(:file) { double("file", save: true) }
+    let(:file) { Yast::Users::SSHAuthorizedKeysFile.new(authorized_keys_path) }
     let(:ssh_dir) { File.join(home, ".ssh") }
     let(:key) { "ssh-rsa 123ABC" }
     let(:authorized_keys_path) { File.join(home, ".ssh", "authorized_keys") }
 
-    before { FileUtils.mkdir_p(home) }
+    before do
+      FileUtils.mkdir_p(ssh_dir)
+
+      allow(Yast::Users::SSHAuthorizedKeysFile).to receive(:new).and_return(file)
+    end
+
     after { FileUtils.rm_rf(tmpdir) if File.exist?(tmpdir) }
 
-    context "if no keys are registered for the given home" do
+    context "when there are not registered keys for the given home" do
       it "does not try to write the keys" do
         expect(file).to_not receive(:save)
+
         keyring.write_keys
       end
     end
 
-    context "if some keys are registered for the given home" do
+    context "when there are registered keys for the given home" do
       let(:uid) { 1001 }
       let(:gid) { 101 }
       let(:home_dir_exists) { true }
       let(:ssh_dir_exists) { false }
 
       before do
-        allow(Yast::SCR).to receive(:Execute).and_call_original
         allow(Yast::SCR).to receive(:Read).and_call_original
+        allow(Yast::SCR).to receive(:Execute).and_call_original
+        allow(Yast::SCR).to receive(:Execute)
+          .with(Yast::Path.new(".target.remove"), authorized_keys_path)
```

Comment:
why mock removal of temporary auth keys path?

Reply:
I have to re-checked it, maybe a leftover (not sure). I was changing the test a lot. But as said, if it does not hurt I prefer to keep both branches _in sync_ as much as possible.